### PR TITLE
Replace Progress Table Library

### PR DESCRIPTION
**Pull request by cforkish**
This PR is the culmination of all the work refactoring our section progress tables to improve performance. The initial task for improving performance was to investigate whether replacing the `react-virtualized` library were were using to render the tables with something more performant. The investigation revealed that the library itself wasn't the problem, but that the whole component tree needed to be refactored to be lighter weight and quicker to render. However, since the `MultiGrid` component we were using from `react-virtualized` specialized in rendering individual cells, we decided the refactor would be cleaner (and hopefully inherently more performant, though this was never explicitly tested once the choice of library was determined to not be the root cause of poor performance) if we swapped out `react-virtualized` for `reactabuler`, which we are already using for most of the tables in our codebase.

All of the refactored components have already been merged, so this PR represent the fairly simple updates to `SectionProgress` to swap out the old table implementations with the new.

### Deployment strategy
I expect this PR to result in a number of diffs in our applitools eyes tests related to subtle changes in the design of the refactored components. I will coordinate with the DOTD on the day that we merge this PR to be on deck to review any such issues.

I will also let Josh Schulte know when it is deployed to be on the lookout for any related Zendesk tickets.

### Future work
`reactabular` as a framework has reached its end of life. there are many new and improved table libraries, but unfortunately they all depend on react 16. so once we upgrade to react 16, we will need to (and _get_ to) start thinking about updating these tables once again. one of the design goals in this refactor was to create a more modular architecture that could be easily be adapted to a new table library, so it is my hope that this work will be relatively easy as compare to this refactor.

on a more immediate note, this PR will be shortly followed by [LP-1744] and [LP-1745](https://codedotorg.atlassian.net/browse/LP-1745) to remove the obsolete code and cleanup directory structure and component naming.

## Links
- [spec](https://docs.google.com/document/d/1SS1DYiLObiccRrVTo3LuLSe5OFiWla2qhsIWlfMvPIo/edit#)
- [jira](https://codedotorg.atlassian.net/browse/LP-1606)

## Testing story
unit tests are passing, but i expect to see some failing ui tests on this build, plus the aforementioned applitools diffs that will need to be resolved. tests for all the new components used by this PR have already been developed and merged.

[LP-1744]: https://codedotorg.atlassian.net/browse/LP-1744

### CODE REVIEW

File: apps/src/templates/sectionProgress/SectionProgress.jsx [hunk 1]
```diff
@@ -127,56 +124,9 @@ class SectionProgress extends Component {
       },
       {includeUserId: true}
     );
-  };
```
Comment (cforkish) [inline apps/src/templates/sectionProgress/SectionProgress.jsx:130]: tooltips are now rendered in their respective components, e.g. `ProgressTableLessonNumber` and `ProgressTableStudentList`

File: apps/src/templates/sectionProgress/SectionProgress.jsx [hunk 2]
```diff
@@ -282,12 +229,7 @@ export default connect(
     currentView: state.sectionProgress.currentView,
     scriptData: getCurrentScriptData(state),
     isLoadingProgress: state.sectionProgress.isLoadingProgress,
-    showStandardsIntroDialog: !state.currentUser.hasSeenStandardsReportInfo,
-    studentTimestamps:
```
Comment (cforkish) [inline apps/src/templates/sectionProgress/SectionProgress.jsx:286]: these props were only needed for the tooltips, which have been relocated per the above comment.

File: apps/src/templates/sectionProgress/SectionProgress.jsx [hunk 3]
```diff
@@ -246,23 +196,20 @@ class SectionProgress extends Component {
               className="fa-pulse fa-3x"
             />
           )}
-          {levelDataInitialized && (
+          {levelDataInitialized && currentView === ViewType.SUMMARY && (
```
Comment (cforkish) [inline apps/src/templates/sectionProgress/SectionProgress.jsx:199]: the astute observer will note that we previously were rendering all three tables at once, regardless of which was visible. we get a nice extra performance boost just with this simple change!

**Review by jamescodeorg — APPROVED** [review]
Looks like a simple straightforward change.  ;-)

congrats
(https://media.giphy.com/media/xT0xezQGU5xCDJuCPe/giphy.gif)

**Comment by cforkish** [comment]
I would like to call out a special thanks to @jamescodeorg for teaching me the idea of pre-merging components to make a "huge" change like this in such a simple PR!

**Review by maureensturgeon — COMMENTED** [review]
Looking at the tooltip changes reminded me that there are bugs with the tooltips not rendering for the right lessons
Screen Shot 2021-02-10 at 10 57 45 AM
Screen Shot 2021-02-10 at 10 58 13 AM

**Comment by cforkish** [comment]
great catch @maureensturgeon! turns out the `relative_position` number we display in the header is not unique within a lesson, so i switched to using `lesson.id` for the tooltip ids.

**Review by maureensturgeon — APPROVED** [review]

**Pull request merged**